main.tf

```diff
@@ -4,6 +4,13 @@ locals {
 
 data "azurerm_client_config" "current" {}
 
+provider "azurerm" {
+  features {}
+  skip_provider_registration = true
+  alias                      = "aks_subscription"
+  subscription_id            = var.aks_subscription_id
+}
```

Comment:
don't include a provider inside of a module. it makes it very difficult to remove it later. 

see the docs at: https://developer.hashicorp.com/terraform/language/modules/develop/providers
while they still allow this syntax they don't allow any modern terraform features to use it

You should instead modify configuration_aliases to require this module is passed in:
https://github.com/hmcts/terraform-module-servicebus-namespace/blob/1b9bd99b936710ab63aeb89c167266f2ad0b09ba/providers.tf#L6